### PR TITLE
i18n(it): Added site-search.mdx

### PR DESCRIPTION
**Pull request by casungo**
#### Description

Translated `site-search.mdx` to Italian

### CODE REVIEW

File: docs/src/content/docs/it/guides/site-search.mdx [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+Di default, i siti Starlight includono la ricerca a testo intero potenziata da [Pagefind](https://pagefind.app/), che è uno strumento di ricerca veloce e a bassa larghezza di banda per i siti statici. Non è richiesta alcuna configurazione per abilitare la ricerca. Compila e distribuisci il tuo sito, quindi usa la barra di ricerca nell'intestazione del sito per trovare i contenuti.
+
+Per escludere una pagina dall'indice di ricerca, aggiungi [`pagefind: false`](/it/reference/frontmatter/#pagefind) al frontmatter della pagina:
+
+## Nascondere contenuti dai risultati di ricerca
+
+### Escludere una pagina
+
+Per escludere una pagina dall'indice di ricerca, aggiungi [`pagefind: false`](/reference/frontmatter/#pagefind) nel frontmatter della pagina:
```
Comment (HiDeoo) [inline docs/src/content/docs/it/guides/site-search.mdx:16]:
```suggestion
Per escludere una pagina dall'indice di ricerca, aggiungi [`pagefind: false`](/it/reference/frontmatter/#pagefind) nel frontmatter della pagina:
```

This should link to the Italian version.

File: docs/src/content/docs/it/guides/site-search.mdx [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+```md title="src/content/docs/partially-indexed.md" ins="data-pagefind-ignore"
+---
+title: Pagina indicizzata parzialmente
+---
+
+Questo testo sarà ricercabile.
+
+<div data-pagefind-ignore>
+
+  Questo testo sarà nascosto dalla ricerca.
```
Comment (HiDeoo) [inline docs/src/content/docs/it/guides/site-search.mdx:40]:
```suggestion
Questo testo sarà nascosto dalla ricerca. 
```

Small change to match the original version.

File: docs/src/content/docs/it/guides/site-search.mdx [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+     ],
+   });
+   ```
+   
+Con questa configurazione aggiornata, la barra di ricerca nel tuo sito aprirà ora una schermata Algolia invece della schermata di ricerca predefinita.
+
+#### Traduzione dell'interfaccia utente di DocSearch
+
+DocSearch fornisce solo stringhe dell'interfaccia utente in inglese di default.
+Aggiungi traduzioni della UI della schermata per la tua lingua utilizzando il sistema di internazionalizzazione integrato in Starlight.
```
Comment (HiDeoo) [inline docs/src/content/docs/it/guides/site-search.mdx:110]: This sentence is missing a link to `/it/guides/i18n/#tradurre-linterfaccia-starlight`, I think it should be around `sistema di internazionalizzazione` but I prefer to let you add it at the proper place.

**Review by HiDeoo — DISMISSED** [review]
Thanks for the translation :raised_hands:

As a non-native speaker, I reviewed this PR using the [Docs Translation Review Checklist](https://hackmd.io/1KYnrpUQR0SAGTzHIUoUDg?view).

I left a few comments and suggestions, nothing major and except for that, everything else looks good to me 👍

**Comment by casungo** [comment]
Thanks for the review! I pushed a commit now, that should fix all of the mistakes pointed out :)

**Review by delucis — COMMENTED** [review]
Thank you for the translation @casungo! It looks good to me. Our link checker is reporting one error for the `#plugins` fragment link in the configuration reference, but that is added in #1452, so this should be good to merge after that.

**Review by delucis — APPROVED** [review]
Link checker passes now that #1452 is merged and I updated the branch 🥳 

Thanks again @casungo.

**Pull request merged**